pkg/controller/association/secret.go

```diff
@@ -0,0 +1,181 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package association
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/jsonpath"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/certificates"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/client"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/net"
+)
+
+// authPasswordUnmanagedSecretKey is the name of the key for the password when using a secret to reference an unmanaged resource
+const authUsernameUnmanagedSecretKey = "username"
+const authPasswordUnmanagedSecretKey = "password"
+
+func (r *Reconciler) ReconcileUnmanagedAssociation(association commonv1.Association) (commonv1.AssociationConf, error) {
+	assocRef := association.AssociationRef()
+	info, err := GetUnmanagedAssociationConnectionInfoFromSecret(r.Client, assocRef)
+	if err != nil {
+		return commonv1.AssociationConf{}, err
+	}
+
+	var ver string
+	ver, err = r.ReferencedResourceVersion(r.Client, r.Parameters.Dialer, assocRef)
```

Comment:
Stupid me. I made this only because of the e2e test. So, I need to revisit the e2e test. One idea is to use a public URL through a nodePort service or a loadBalancer (which means limiting the test to cloud providers that offer it).

Reply:
I updated the e2e test to use a nodePort. The downside to this is that it involves to open the node port but since the communications are encrypted, there is authentication, and it's only for a test cluster, I think that's ok.